test/hotspot/jtreg/runtime/cds/appcds/ClassPathAttr.java

```diff
@@ -126,6 +128,52 @@ static void testNonExistentJars() throws Exception {
         });
   }
 
+  static void testClassPathAttrJarOnCP() throws Exception {
+    String helloJar = JarBuilder.getOrCreateHelloJar();
+    String jar1 = TestCommon.getTestJar("cpattr1.jar");
+    String cp = jar1 + File.pathSeparator + helloJar;
+
+    // The cpattr1.jar contains "Class-Path: cpattr2.jar".
+    // The cpattr2.jar contains "Class-Path: cpattr3.jar cpattr5_123456789_223456789_323456789_42345678.jar".
+    // With -cp cpattr1:hello.jar, the following shared paths should be stored in the CDS archive:
+    // cpattr1.jar:cpattr2.jar:cpattr3.jar:cpattr5_123456789_223456789_323456789_42345678.jari:hello.jar
+    TestCommon.testDump(cp, TestCommon.list("Hello"),
+        "-Xlog:class+path");
+
+    // Run with the same -cp apattr1.jar:hello.jar. The Hello class should be
+    // loaded from the archive.
+    TestCommon.run(
+        "-Xlog:class+path,class+load",
+        "-cp", cp,
+        "Hello")
+      .assertNormalExit(output -> {
+          output.shouldContain("Hello source: shared objects file");
+        });
```

Comment:
Style nit here and elsewhere:
```suggestion
    TestCommon.run("-Xlog:class+path,class+load",
                   "-cp", cp,
                   "Hello")
               .assertNormalExit(output -> {
                     output.shouldContain("Hello source: shared objects file");
                 });
```